security/vTPM-tests.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM.
+# Author: Nageswara R Sastry <rnsastry@linux.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import dmesg, genio, linux_modules
+
+
+class PerfBasic(Test):
+    """
+    vTPM tests for Linux
+    :avocado: tags=privileged,security,tpm
+    """
+    def setUp(self):
+        device_tree_path = "/proc/device-tree/vdevice/vtpm@30000003"
+        if not os.path.exists(device_tree_path):
+            self.cancel("vTPM not enabled.")
+        compatible_file = "%s/compatible" % device_tree_path
+        if os.path.exists(compatible_file):
+            self.cvalue = genio.read_file(compatible_file).rstrip('\t\r\n\0')
+            self.cvalue = self.cvalue.split(",")[1]
+            self.log.info("TPM version is %s" % self.cvalue)
+        else:
+            self.fail("Can't determine version of TPM")
+        self.no_config = []
+        self.no_device = []
+
+    def test_TPM_register(self):
+        output = dmesg.collect_errors_dmesg('tpm_ibmvtpm 30000003: CRQ initialization completed')
```

Comment:
is this should be individual test or need to part of per-requist of every test ?, if not can we have  info message as `Tpm device register`

Reply:
Can be individual test.

Reply:
sure in that case if we hit find string  dmesg we should be have info message to user as we have device register

Reply:
Sure, added the message. Thanks!!